### PR TITLE
refactor(a2a): merge slug into path with automatic normalization

### PR DESCRIPTION
**Pull request by StefanMaker**
Consolidate A2AAgent.slug and A2AAgent.path into a single normalized path field. The path now automatically converts inputs with slashes

### CODE REVIEW

File: scripts/migrate_a2a_agent_path_slug.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python3
+"""
+Data Migration Script: Merge A2AAgent.slug into A2AAgent.path
+
+This script migrates the A2AAgent collection by:
+1. Detecting and reporting any path conflicts before migration
+2. Converting all path values to slug format (no slashes)
+3. Removing the slug field from all documents
```
Comment (kxue43) [inline scripts/migrate_a2a_agent_path_slug.py]: This is change request [m1].

The 4-step docstring here is out-of-sync with the actual code:
- Step 3 and 4 are in the reverse order compared to what happens in the code.
- The code has 5 steps, while here only 4 are listed.

Let's update the docstring here for future readers.

File: scripts/migrate_a2a_agent_path_slug.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    if not dry_run:
+        logger.info("Removing 'slug' field from all documents...")
+        result = await collection.update_many({"slug": {"$exists": True}}, {"$unset": {"slug": ""}})
+        stats["slug_fields_removed"] = result.modified_count
+        logger.info(f"Removed 'slug' field from {stats['slug_fields_removed']} documents")
+
+    return stats
+
+
+async def main(dry_run: bool = False, force: bool = False) -> None:
```
Comment (kxue43) [inline scripts/migrate_a2a_agent_path_slug.py]: This is change request [m2].

The parameter `force` is never used in the body of `main`. Let's remove it here and at the call site.

File: registry-pkgs/src/registry_pkgs/models/a2a_agent.py [hunk 3]
```diff
@@ -94,6 +94,21 @@
 }
 
 
+def normalize_a2a_agent_path(value: Any) -> str:
```
Comment (kxue43) [inline registry-pkgs/src/registry_pkgs/models/a2a_agent.py:97]: This is change request [m4].

`normalize_a2a_agent_path` is the most important logic change in this PR and it's a pure function with no side effects — ideal to unit-test. It currently has zero direct test coverage.

Please add tests covering at minimum the four cases listed in the `_normalize_path` docstring, plus these additional edge cases:

| Input | Expected output | Why it matters |
|---|---|---|
| `None` | raises `ValueError` | explicit `None` guard |
| `"/"` | raises `ValueError` | root-path rejection |
| `""` | raises `ValueError` | empty input |
| `"   "` | raises `ValueError` | whitespace-only (stripped to empty) |
| `"___"` | raises `ValueError` | all non-alphanumeric, nothing survives regex |
| `"deep-intel"` | `"deep-intel"` | idempotency — critical for migration re-runnability |
| `"MyAgent"` | `"myagent"` | uppercase folding |
| `"my_agent"` | `"my-agent"` | underscores |
| `"my.agent"` | `"my-agent"` | dots |
| `"/Team A/CRM Agent/"` | `"team-a-crm-agent"` | realistic multi-segment path input |
| `"café"` | `"caf"` | Unicode silently stripped (`é` is not in `[a-z0-9]`, becomes a dash, gets stripped) — no error raised; worth at least a test or comment marking this as intentional |

File: registry-pkgs/tests/unit/workflow/test_executor_resolver.py [hunk 4]
```diff
@@ -156,7 +156,7 @@ async def test_resolve_executor_supports_builtin_echo_without_db_lookup(self, mo
     async def test_resolve_executor_falls_back_to_a2a_agent(self, monkeypatch: pytest.MonkeyPatch):
         self._patch_beanie_filters(monkeypatch)
         monkeypatch.setattr(executor_resolver.ExtendedMCPServer, "find_one", AsyncMock(return_value=None))
-        monkeypatch.setattr(executor_resolver.A2AAgent, "find_one", AsyncMock(return_value=_a2a_agent("/deep-intel")))
+        monkeypatch.setattr(executor_resolver.A2AAgent, "find_one", AsyncMock(return_value=_a2a_agent("deep-intel")))
```
[the remark this answers is not in the document]
Comment (kxue43) [reply registry-pkgs/tests/unit/workflow/test_executor_resolver.py:159]: This is just a note for future reference.

The test is fine as-is, but this comment reveals that the changes in this PR breaks the existing code around workflows.

I have created another ticket [AS-1633](https://ascending.atlassian.net/browse/AS-1633) to fix these breaks. Let's work on AS-1633 once this PR is merged.

File: registry/src/registry/services/a2a_agent_service.py [hunk 5]
```diff
@@ -316,24 +325,13 @@ async def get_agent_by_path(self, path: str) -> A2AAgent | None:
...
-
-        Args:
-            slug: Agent slug used in proxy routes (e.g., deep-intel)
+            path: Registry path or path-like input (e.g., /deep-intel)
 
         Returns:
             Agent document, or None if not found
         """
-        return await A2AAgent.find_one({"slug": slug})
+        normalized_path = normalize_a2a_agent_path(path)
```
Comment (kxue43) [inline registry/src/registry/services/a2a_agent_service.py]: This is change request [m5].

`get_agent_by_path` normalizes its input before querying, which silently redirects a semantically distinct path to an existing slug. For example, a caller that passes `"deep/intel"` would be silently resolved to `"deep-intel"` if that slug exists — masking an upstream bug rather than surfacing it.

Post-migration, `A2AAgent.path` stores bare slugs enforced by the model validator at every write. Any caller passing a value that isn't already a valid slug signals a problem at the call site, not a normalization opportunity here. Remove the `normalize_a2a_agent_path` call and query directly:

```python
async def get_agent_by_path(self, path: str) -> A2AAgent | None:
    return await A2AAgent.find_one({"path": path})
```

File: registry/tests/fixtures/factories.py [hunk 6]
```diff
@@ -15,7 +15,7 @@ class ServerInfoFactory(factory.DictFactory):
 
     server_name = factory.LazyFunction(lambda: fake.company())
     description = factory.LazyFunction(lambda: fake.text(max_nb_chars=200))
-    path = factory.LazyFunction(lambda: f"/{fake.slug()}")
+    path = factory.LazyFunction(lambda: fake.slug())  # Path is now in slug format (no slashes)
```
[the remark this answers is not in the document]
Comment (kxue43) [reply registry/tests/fixtures/factories.py]: This is change request [m6].

I agree with Copilot. Let's revert this. `ServerInfoFactory` is not used in agent test cases anyway, and we don't need to add a new agent factory class.

**Comment by kxue43** [comment]
@StefanMaker 

Overall the PR looks good. I made a few minor change requests. Once addressed, this PR is ready for merging.

While reviewing this PR, I noticed some related bugs in both frontend (AS-1632) and backend (AS-1633) around workflow's use of `A2AAgent.path`. Once this PR is merged and data migrated, let's start AS-1633 immediately.

**Review by kxue43 — APPROVED** [review]
LGTM. Thanks for the good work!
